plugins/roadie-backstage-entity-validator/src/validator.ts

```diff
@@ -16,21 +16,22 @@ import {
   resourceEntityV1alpha1Validator,
   entityKindSchemaValidator,
   makeValidator,
+  Entity,
 } from '@backstage/catalog-model';
 import { templateEntityV1beta3Validator } from '@backstage/plugin-scaffolder-common';
 import annotationSchema from './schemas/annotations.schema.json';
 import repositorySchema from './schemas/repository.schema.json';
 import productSchema from './schemas/product.schema.json';
-import Ajv from 'ajv';
+import Ajv, { ValidateFunction } from 'ajv';
 import ajvFormats from 'ajv-formats';
 import { relativeSpaceValidation } from './relativeSpaceValidation';
 
 const ajv = new Ajv({ verbose: true });
 ajvFormats(ajv);
 
-function ajvCompiledJsonSchemaValidator(schema) {
+function ajvCompiledJsonSchemaValidator(schema: any) {
   return {
-    async check(data) {
+    async check(data: any) {
       return entityKindSchemaValidator(schema)(data) === data;
```

Comment:
Not your issue... Im thinking out loud.

We are effectively checking the schemas are schemas at runtime here.. if they arent schemas im assuming it will fail here. maybe it makes sense to convert those json import files to ts with type checking..